src/main/java/org/bookkeepervisualmanager/bookkeeper/BookkeeperManager.java

```diff
@@ -79,11 +91,37 @@ public BookkeeperManager(ConfigurationStore configStore) throws BookkeeperExcept
             LOG.log(Level.INFO, "Starting bookkeeper admin.");
             this.bkAdmin = new BookKeeperAdmin(bkClient);
 
-        } catch (Throwable t) {
+            this.metadataCache = metadataCache;
+        } catch (IOException | InterruptedException | BKException | ConfigurationNotValidException t) {
             throw new BookkeeperException(t);
         }
     }
 
+    public void refreshMetadataCache() throws BookkeeperException {
+        LOG.info("Refreshing Metadata Cache");
+        try {
+            Iterable<Long> ledgersIds = bkAdmin.listLedgers();
+            for (long ledgerId : ledgersIds) {
+                LedgerMetadata ledgerMetadata = readLedgerMetadata(ledgerId);
+                Ledger ledger = new Ledger(ledgerId,
+                        ledgerMetadata.getLength(),
+                        new java.sql.Timestamp(ledgerMetadata.getCtime()),
+                        new java.sql.Timestamp(System.currentTimeMillis()),
+                        Base64.getEncoder().encodeToString(serDe.serialize(ledgerMetadata)));
```

Comment:
ZK metadata is serialized to a byte[] by the native BK API (serDe), then we are storing it in base64 enconding on DB